src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+        } else {
+            size = addEntry(Objects.hashCode(key), key, value, getIndex(key)) ? ++size : size;
+        }
+        resize();
```

Comment:
Let's not call the resize() method every time. Let's add a check here.
```suggestion
        if (size >= threshold) {
           resize();
        }
```